diff -up systemtap-4.7/python/HelperSDT/_HelperSDT.c.omv~ systemtap-4.7/python/HelperSDT/_HelperSDT.c
--- systemtap-4.7/python/HelperSDT/_HelperSDT.c.omv~	2022-09-26 21:15:43.686689832 +0200
+++ systemtap-4.7/python/HelperSDT/_HelperSDT.c	2022-09-26 21:16:19.271903219 +0200
@@ -13,6 +13,7 @@
 
 // PR25841: ensure that the libHelperSDT.so file contains debuginfo
 // for the tapset helper functions, so they don't have to look into libpython*
+#include <internal/pycore_frame.h> // PyFrameObject definition
 #include <frameobject.h>
 PyFrameObject _dummy_frame;
 #include <object.h>
