diff -Naur systemtap-2.0/rpm_finder.cxx systemtap-2.0.tpg/rpm_finder.cxx
--- systemtap-2.0/rpm_finder.cxx	2012-10-10 00:08:59.000000000 +0000
+++ systemtap-2.0.tpg/rpm_finder.cxx	2013-02-02 15:00:14.873783988 +0000
@@ -24,7 +24,10 @@
 
 #define _RPM_4_4_COMPAT
 #include <string.h>
-#include <rpm/rpmlib.h>
+#include <rpm/rpmevr.h>
+#include <rpmtypes.h>
+#include <rpm/rpmiotypes.h>
+#include <rpm/rpmpgp.h>
 #include <rpm/rpmts.h>
 #include <rpm/rpmdb.h>
 #include <rpm/header.h>
