--- systemtap-3.1/configure.ac.0002~	2017-10-04 12:15:32.137603452 +0200
+++ systemtap-3.1/configure.ac	2017-10-04 12:17:43.696189276 +0200
@@ -464,25 +464,14 @@ dnl Look for librpm.
 AC_ARG_WITH([rpm],
   [AS_HELP_STRING([--with-rpm],
                   [query rpm database for missing debuginfos])], [], [with_rpm="auto"])
-if test "$with_rpm" != "no"; then
-  AC_CHECK_LIB(rpm, rpmtsInitIterator, [
-                    AC_DEFINE([HAVE_LIBRPM],[1],[have librpm])
-                    stap_LIBS="$stap_LIBS -lc -lrpm"
-                    have_librpm="yes"], [have_librpm="no"])
-  dnl explicit -lrpmdb is a separate requirement on some older librpms
-  AC_CHECK_LIB(rpmdb, rpmdbNextIterator, [
-                    stap_LIBS="$stap_LIBS -lrpmdb"])
-  AC_CHECK_LIB(rpmio, rpmFreeCrypto, [
-                    AC_DEFINE([HAVE_LIBRPMIO],[1],[have librpmio])
-                    stap_LIBS="$stap_LIBS -lc -lrpmio"
-                    have_librpmio="yes"], [have_librpmio="no"])
-  if test "x$have_librpm" != "xyes" -a "$with_rpm" == "yes"; then
-     AC_MSG_ERROR([cannot find librpm])
-  fi
-  if test "x$have_librpmio" != "xyes" -a "$with_rpm" == "yes"; then
-     AC_MSG_WARN([cannot find librpmio])
-  fi
-fi
+AS_IF([test "x$with_rpm" != "xno"], [
+  PKG_CHECK_MODULES([rpm], [rpm],
+    [have_rpm=yes
+     AC_DEFINE([HAVE_LIBRPM], [1], [Define to 1 if you have the rpm libraries.])
+    ], [have_rpm=no])
+], [have_rpm=no])
+
+AM_CONDITIONAL([HAVE_LIBRPM], [test "${have_rpm}" = "yes"])
 
 dnl Look for readline.
 dnl
--- systemtap-3.1/Makefile.am.0002~	2017-02-17 18:37:01.000000000 +0100
+++ systemtap-3.1/Makefile.am	2017-10-04 12:15:32.136603455 +0200
@@ -128,6 +128,12 @@ stap_CPPFLAGS += $(avahi_CFLAGS)
 stap_LDADD += $(avahi_LIBS)
 endif
 
+if HAVE_LIBRPM
+stap_CXXFLAGS += $(rpm_CFLAGS)
+stap_CPPFLAGS += $(rpm_CFLAGS)
+stap_LDADD += $(rpm_LIBS)
+endif
+
 if HAVE_NSS
 stap_SOURCES += nsscommon.cxx csclient.cxx cscommon.cxx 
 stap_CFLAGS += $(nss_CFLAGS) -DSTAP
--- systemtap-3.1/rpm_finder.cxx.0002~	2017-02-17 18:37:01.000000000 +0100
+++ systemtap-3.1/rpm_finder.cxx	2017-10-04 12:15:32.137603452 +0200
@@ -24,10 +24,6 @@ extern "C" {
 
 #define _RPM_4_4_COMPAT
 #include <string.h>
-#include <rpm/rpmlib.h>
-#include <rpm/rpmts.h>
-#include <rpm/rpmdb.h>
-#include <rpm/header.h>
 
 #ifndef xfree
 #define xfree free
@@ -35,6 +31,8 @@ extern "C" {
 
 }
 
+#include <rpm/rpm4compat.h>
+
 #if ! HAVE_LIBRPMIO && HAVE_NSS
 extern "C" {
 #include <nss.h>
